py/stabilizer/stream.py

```diff
@@ -87,19 +88,31 @@ class StabilizerStream(asyncio.DatagramProtocol):
     }
 
     @classmethod
-    async def open(cls, local_addr, maxsize=1):
+    async def open(cls, addr, port, broker, maxsize=1):
         """Open a UDP socket and start receiving frames"""
         loop = asyncio.get_running_loop()
-        transport, protocol = await loop.create_datagram_endpoint(
-            lambda: cls(maxsize), local_addr=local_addr)
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
         # Increase the OS UDP receive buffer size to 4 MiB so that latency
         # spikes don't impact much. Achieving 4 MiB may require increasing
         # the max allowed buffer size, e.g. via
         # `sudo sysctl net.core.rmem_max=26214400` but nowadays the default
         # max appears to be ~ 50 MiB already.
-        sock = transport.get_extra_info("socket")
-        if sock is not None:
-            sock.setsockopt(socket.SOL_SOCKET, socket.SO_RCVBUF, 4 << 20)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_RCVBUF, 4 << 20)
+
+        # We need to specify which interface to receive broadcasts from, or Windows may choose the
+        # wrong one. Thus, use the broker address to figure out our local address for the interface
+        # of interest.
+        if ipaddress.ip_address(addr).is_multicast:
+            print('Subscribing to multicast')
+            group = socket.inet_aton(addr)
+            iface = socket.inet_aton('.'.join([str(x) for x in get_local_ip(broker)]))
+            sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, group + iface)
+
+        sock.bind(('', port))
```

Comment:
```suggestion
            sock.bind(('', port))
        else:
            sock.bind((addr, port))
```

Reply:
Already added in ae66208487337bc7ae3e503cf2c69888b1c3ebdb